README.md

```diff
@@ -1,13 +1,203 @@
+| `1.2.3-beta` > `1.2.3-1` | Where strings and integers must be compared, strings have a higher precedence than integers. |
+| `1.2.3-beta+abcd` = `1.2.3-beta` | Build-metadata is not to be used when comparing. |
+| `1.2.3-beta+sha.0beef` = `1.2.3-beta+sha.80081` | Build-metadata is not to be used when comparing. |
+
+## Library Usage
+#### Creating the object directly:
+Imports the library, and creates a `SemanticVersion` object directly.
 ```toit
-import semver
+import semver show *
 
 main:
-  a := "1.0.0"
-  b := "1.0.0-beta.1"
-  print (semver.compare a b)  // => 1
+  // Instantiation by direct creation
+  semver-a := SemanticVersion 1 2 3
+
+  // Prints 1.2.3
+  print "$semver-a"
+
+  // Instantiation by direct creation, without 'minor'
+  semver-b := SemanticVersion 1 2
+
+  // Prints 1.2.0
+  print "$semver-b"
+
+  // Instantiation by direct creation, without 'minor' or 'patch'
+  semver-c := SemanticVersion 1
+
+  // Prints 1.0.0
+  print "$semver-c"
+```
+#### Directly creating including pre-release:
+Pre-release and build-metadata can also be specified directly.  These are held
+as a list and must be specified as a list, even if only one element:
+```toit
+  // (Contiues from previous example)
+
+  // Direct instantiation.
+  semver-d := SemanticVersion 1 0 0 ["alpha","1"] ["sha",23132]
+
+  // Prints 1.0.0-alpha.1+sha.23132
+  print "$semver-d"
+```
+#### Immutability:
+Since the object is immutable, editing one of the fields after creation is not
+possible.
+```toit
+  // (Contiues from previous examples)
+
+  // Direct instantiation including pre-release.
+  semver-e := SemanticVersion 1 2 3 ["alpha",1]
+
+  // Prints 2
+  print "$(semver-e.minor)"
+
+  // Fails/throws
+  //semver-e.minor = 5
+```
+#### Object instantiation by string parsing:
+The library parses strings into a `SemanticVersion` object, which has methods
+and functions.  Comparison operators are shown in the example below.
+```toit
+  // (Contiues from previous examples)
+
+  // strings
+  string-f := "1.0.0"
+  string-g := "1.0.0-beta.1"
+
+  // parse strings into SemanticVersion objects
+  semver-f := SemanticVersion.parse string-f
+  semver-g := SemanticVersion.parse string-g
+
+  // compare two objects: prints "f is later than g."
+  if semver-f > semver-g:
+    print "f is later than g."
+  else:
+    print "g is later than f."
+
+  // compare two objects: prints "f and g are different."
+  if semver-f == semver-g:
+    print "a and b are the same."
+  else:
+    print "a and b are different."
+
+  // compare two objects: prints "f and f are the same."
+  if semver-f == semver-f:
+    print "f and f are the same."
+  else:
+    print "f and f are different."
+
 ```
+
+#### Simple comparison using strings only:
+This is implemented, but not recommended as it is a computationally expensive -
+string parsing happens every time an evaluation is made. (Code uses the library
+and creates/destroys objects in the background):
```

Comment:
```suggestion
For convenience and backwards compatibility, it is also possible to compare strings directly. In
the background the library creates the corresponding `SemanticVersion` instances and uses
them for comparison.

Similar to all `compare-to` functions the `compare` function returns -1 if the left-hand side is less
than the right-hand side; 0 if they are equal, and 1 otherwise.
```